### PR TITLE
feat(v2): add scripts and stylesheets field to config

### PR DESCRIPTION
**Pull request by endiliey**
## Motivation

 Add `scripts` and `stylesheets` field to docusaurus.config. for v1 parity and enable user to use external resources script or stylesheets

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/Docusaurus/blob/master/CONTRIBUTING.md#pull-requests)?

yes

## Test Plan

Add random scripts and stylesheets into config
```js
  scripts: [
    'https://docusaurus.io/slash.js',
    {
      src:
        'https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.0/clipboard.min.js',
      async: true,
    },
  ],
  stylesheets: [
    'https://docusaurus.io/style.css',
    {
      href: 'http://css.link',
      type: 'text/css',
    },
  ],
```

### CODE REVIEW

File: packages/docusaurus/src/client/App.js [hunk 1]
```diff
@@ -16,8 +17,29 @@ import PendingNavigation from './PendingNavigation';
...
+        <Head>
+          {stylesheets &&
+            stylesheets.map(source =>
+              source.href ? (
+                <link rel="stylesheet" key={source.href} {...source} />
+              ) : (
+                <link rel="stylesheet" key={source} href={source} />
+              ),
+            )}
+          {scripts &&
```
Comment (yangshun) [inline packages/docusaurus/src/client/App.js]: Do we inject scripts here in V1? Putting scripts here blocks page rendering which is bad, unless they use `async`. Maybe we should allow injecting before `</body>`, but that'll be more complicated.

File: packages/docusaurus/src/client/App.js [hunk 2]
```diff
@@ -16,8 +17,29 @@ import PendingNavigation from './PendingNavigation';
...
 
 function App() {
+  const {stylesheets, scripts} = siteConfig;
   return (
     <DocusaurusContext.Provider value={{siteConfig}}>
+      {(stylesheets || scripts) && (
+        <Head>
+          {stylesheets &&
+            stylesheets.map(source =>
+              source.href ? (
```
Comment (yangshun) [inline packages/docusaurus/src/client/App.js]: Prefer to use `typeof source === 'string'` to do the check.

File: packages/docusaurus/src/client/App.js [hunk 3]
```diff
@@ -16,8 +17,29 @@ import PendingNavigation from './PendingNavigation';
...
+                <link rel="stylesheet" key={source.href} {...source} />
+              ) : (
+                <link rel="stylesheet" key={source} href={source} />
+              ),
+            )}
+          {scripts &&
+            scripts.map(source =>
+              source.src ? (
```
Comment (yangshun) [inline packages/docusaurus/src/client/App.js]: Ditto

File: packages/docusaurus/src/client/App.js (hunk 1 above)
Comment (endiliey) [reply packages/docusaurus/src/client/App.js]: yep its done like this in v1

**Comment by CMCDragonkai** [comment]
I followed the SO link here: https://stackoverflow.com/a/58349826/582917.

Is it possible to add in custom HTML tags like `base` tag into the head section?

**Comment by Josh-Cena** [comment]
@CMCDragonkai Please don't necropost—perfect way to get your question undiscoverable.

We don't have a first-class API for any tag, but you can get away with [a plugin API](https://docusaurus.io/docs/next/api/plugin-methods/lifecycle-apis#injectHtmlTags):

```js
// docusaurus.config.js
module.exports = {
  // ...
  plugins: [
    () => ({
      name: "inject-tag",
      injectHtmlTags() {
        return {
          headTags: [{ tagName: "base", attributes: { href: "https://www.example.com/" } }],
        };
      },
    }),
  ],
};
```

**Pull request merged**